File: python/python/basic.py
import svar

basic=svar.load("/data/zhaoyong/Program/Apps/Tests/JSVar/python/build/libbasic.so")

pyobj={"i":1,"d":1.2,"s":"hello world","v":[1,2,3]}
print("pyobj is:",pyobj)

ret=basic.obtainInfo(pyobj)
print("ret is:",ret)
print("convert to py:",ret.py())

print(basic.dtos(100.423))

c=basic.add(34,3)
print(c)
print(basic.add(c,3))


#robj=basic.obtainInfo(pyobj)
#print(robj)

File: python/src/basic.cpp
#define __SVAR_BUILDIN__
#include "../../src/Svar.h"

GSLAM::Svar obtainInfo(GSLAM::Svar pyobj){
    std::cerr<<"Calling obtainInfo"<<std::endl;
    std::cout<<"Argument is "<<pyobj;
    GSLAM::Svar ret=GSLAM::Svar::object({{"1",2.},{"buildin",std::vector<GSLAM::Svar>({1,2,3})}});
    return ret;
}

std::string dtos(double& d){
    return std::to_string(d);
}

int add(int a,int b){
    return a+b;
}

REGISTER_SVAR_MODULE(basic)
{
    svar.def("obtainInfo",obtainInfo);
    svar.def("dtos",dtos);
    svar.def("add",add);
}
